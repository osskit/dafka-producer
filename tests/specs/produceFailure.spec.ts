import type {Orchestrator} from '../testcontainers/orchestrator.js';
import {start} from '../testcontainers/orchestrator.js';

describe('tests', () => {
    let orchestrator: Orchestrator;

    beforeEach(async () => {
        orchestrator = await start(
            {
                KAFKA_BROKER: 'kafka:9092',
            },
            []
        );
    }, 5 * 60 * 1000);

    afterEach(async () => {
        if (!orchestrator) {
            return;
        }
        await orchestrator.stop();
    });

    it('produce failure', async () => {
        expect(() =>
            orchestrator.dafkaProducer.produce([
                {
                    topic: 'not exists',
                    value: {data: 'foo'},
                },
            ])
        ).rejects.toMatchSnapshot();
    });
});
