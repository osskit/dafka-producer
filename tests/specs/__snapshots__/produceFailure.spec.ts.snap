// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests produce failure 1`] = `[TypeError: fetch failed]`;
